NEWS.md

```diff
@@ -1,9 +1,13 @@
 # teal.data 0.3.0.9007
 
+### Enhancements
+* Simplified `join_key` to better support primary keys. Added `primary_key` as a wrapper to `join_key`.
```

Comment:
please add information about `[` and `[<-`

Reply:
I added the following:

> * Added subset and subset-assignment to `JoinKeySet` class to manipulate relationship pair keys _(`[` and `[<-`)_.